sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListenerSuite.scala

```diff
@@ -36,14 +36,23 @@ import org.apache.spark.sql.catalyst.util.quietly
 import org.apache.spark.sql.execution.{LeafExecNode, QueryExecution, SparkPlanInfo, SQLExecution}
 import org.apache.spark.sql.execution.metric.{SQLMetric, SQLMetrics}
 import org.apache.spark.sql.test.SharedSQLContext
-import org.apache.spark.status.config._
+import org.apache.spark.status.config.LIVE_ENTITY_UPDATE_PERIOD
 import org.apache.spark.util.{AccumulatorMetadata, JsonProtocol, LongAccumulator}
 import org.apache.spark.util.kvstore.InMemoryStore
 
-class SQLListenerSuite extends SparkFunSuite with SharedSQLContext with JsonTestUtils {
+
+class SQLAppStatusListenerSuite extends SparkFunSuite with SharedSQLContext with JsonTestUtils {
   import testImplicits._
 
-  override protected def sparkConf = super.sparkConf.set(LIVE_ENTITY_UPDATE_PERIOD, 0L)
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    sparkContext.conf.set(LIVE_ENTITY_UPDATE_PERIOD, 0L)
```

Comment:
only set this config for this suite.

Reply:
I commented on the other PR where you mentioned this, but I still don't get what this is changing. I don't see any global state that is being overriden by `override protected def sparkConf`. This test suite only extends traits (e.g. `SharedSQLContext` which extends `SharedSparkSession`), and those only keep suite-level state, not global state.

There are other tests that do the same thing.